cxx-sensors/src/main/java/org/sonar/cxx/sensors/drmemory/CxxDrMemorySensor.java

```diff
@@ -82,22 +87,39 @@ protected void processReport(final SensorContext context, File report) {
     LOG.debug("Parsing 'Dr Memory' format");
 
     for (DrMemoryError error : DrMemoryParser.parse(report, defaultCharset())) {
+      List<CxxReportLocation> locations = new ArrayList<>();
       if (error.getStackTrace().isEmpty()) {
-        saveUniqueViolation(context, CxxDrMemoryRuleRepository.KEY,
-          null, null,
-          error.getType().getId(), error.getMessage());
+        CxxReportLocation moduleLocation = new CxxReportLocation(null, null, error.getMessage());
+        locations.add(moduleLocation);
+      } else {
+        addFileLocations(context, error, locations);
       }
-      for (Location errorLocation : error.getStackTrace()) {
-        if (isFileInAnalysis(context, errorLocation)) {
-          saveUniqueViolation(context, CxxDrMemoryRuleRepository.KEY,
-            errorLocation.getFile(), errorLocation.getLine().toString(),
-            error.getType().getId(), error.getMessage());
-          break;
-        }
+      if (!locations.isEmpty()) {
+        CxxReportIssue issue = new CxxReportIssue(CxxDrMemoryRuleRepository.KEY, error.getType().getId(), locations);
+        saveUniqueViolation(context, issue);
       }
     }
   }
 
+  private void addFileLocations(final SensorContext context, DrMemoryError error, List<CxxReportLocation> locations) {

```

Comment:
same logic is implemented in CxxValgrindSensor

For each CxxReportLocation CxxReportSensor will check if path belongs to the project or not. If it doesn't - this [secondary] location will be ignored. So finally we will have 

* a primary location which is guaranteed to belong to the project (same as before)
* ```>= 0``` additional locations, which reference files from the analyzed project;